eng/common/scripts/modules/git-api-calls.psm1

```diff
@@ -0,0 +1,81 @@
+# Methods for interacting with the github or azure repo APIs
+# Retrieves the list of all tags that exist on the target repository
+# Changes 1: Pass full API URL instead of formulating it within the method
+# Changes 2: Extend method for git Repos
+function GetExistingTags($apiUrl, $repoType = "github", $headers = $null) {
+    try {
+      if ($repoType -eq "github")
+      {
+        return (Invoke-RestMethod -Method "GET" -Uri $apiUrl  ) | % { $_.ref.Replace("refs/tags/", "") }
```

Comment:
What is unique about github here? Can't this be controlled by the headers parameter?

Reply:
What other repo type are we supporting?

Reply:
Its intended to support only github and Azure git Repos. The Azure Git call requires authentication for getting tags but github does not. Also the returned object is formatted slightly differently between the two repo types.